vendor/go.etcd.io/etcd/client/v3/kubernetes.go

```diff
@@ -0,0 +1,179 @@
+package clientv3
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+)
+
+func NewKubernetes(c *Client) Kubernetes {
+	return &kubernetes{kv: RetryKVClient(c), watcher: c}
+}
+
+type Kubernetes interface {
+	Get(ctx context.Context, key string, revision int64) (kv *mvccpb.KeyValue, rev int64, err error)
+	List(ctx context.Context, prefix string, revision, limit int64, fromKey string) (kvs []*mvccpb.KeyValue, count, rev int64, err error)
+	Count(ctx context.Context, prefix string) (int64, error)
+	OptimisticCreate(ctx context.Context, key string, value []byte) (success bool, rev int64, err error)
+	OptimisticUpdate(ctx context.Context, key string, value []byte, expectedRevision int64) (kv *mvccpb.KeyValue, success bool, rev int64, err error)
+	OptimisticDelete(ctx context.Context, key string, expectedRevision int64) (kv *mvccpb.KeyValue, success bool, rev int64, err error)
+
+	// Consider merging
+	WatchPrefix(ctx context.Context, prefix string, rev int64) WatchChan
+	WatchKey(ctx context.Context, key string, rev int64) WatchChan
```

Comment:
What's the purpose of splitting these two?
I would really prefer to have a single `Watch` method with an additional parameter for isRecursive

Reply:
Why does apiserver establish watch for single object at all? Shouldn't it demultiplex watch calls?

Reply:
Hmm, makes sense. I agree that it makes sense to disable caching etcd state in apiserver in environments with limited memory, however watch demultiplexing seems like a always net positive thing.